react/Paywall/OnlyOfficePaywall.tsx

```diff
@@ -0,0 +1,86 @@
+import React, { FC, useMemo } from 'react'
+import ReactMarkdown from 'react-markdown'
+import get from 'lodash/get'
+
+import { models, useClient } from 'cozy-client'
+
+import Paywall from './Paywall'
+import useInstance from '../helpers/useInstance'
+import withLocales from '../I18n/withLocales'
+import fr from './locales/fr.json'
+import en from './locales/en.json'
+
+const locales = {
+  en,
+  fr
+}
+
+interface OnlyOfficePaywallProps {
+  isPublic: boolean
+  onClose: () => void
+  t: (message: string, options?: object) => string
+}
+
+const { instance: instanceModel } = models
+
+const OnlyOfficePaywall: FC<OnlyOfficePaywallProps> = ({
+  onClose,
+  isPublic,
+  t
+}) => {
+  const client = useClient()
+  const { isLoaded, ...instance } = useInstance(client)
+
+  const arePremiumLinksEnabled = useMemo(
+    () => instanceModel.arePremiumLinksEnabled(instance),
+    [instance]
+  )
+
+  if (!isLoaded) return null
+
+  if (arePremiumLinksEnabled && isPublic) {
+    return (
+      <Paywall
+        title={t('onlyOfficePaywall.public.title')}
+        content={t('onlyOfficePaywall.public.content')}
+        actionLabel={t('onlyOfficePaywall.public.action')}
+        onClose={onClose}
+        onAction={onClose}
+      />
+    )
+  }
+
+  if (arePremiumLinksEnabled) {
+    return (
+      <Paywall
+        title={t('onlyOfficePaywall.premiumLink.title')}
+        content={t('onlyOfficePaywall.premiumLink.content')}
+        actionLabel={t('onlyOfficePaywall.premiumLink.action')}
+        onClose={onClose}
+        onAction={(): void => {
```

Comment:
on préfère les handlers (cf discussion point front)

Reply:
I suggest adding it in cozy-guideline to help remember it in the future : https://github.com/cozy/cozy-guidelines/pull/88